samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -354,7 +355,19 @@ object SamzaContainer extends Logging {
 
     val chooserFactory = Util.getObj[MessageChooserFactory](chooserFactoryClassName)
 
-    val chooser = DefaultChooser(inputStreamMetadata, chooserFactory, config, samzaContainerMetrics.registry)
+    /**
+     * Constructs a map from SystemStreamPartition to number of tasks consuming from it.
+     * To mark a stream as bootstrapped, each SSP in the stream must be completely consumed
+     * by all the tasks that are assigned to it. This will help in enforcing the condition.
+     */
+    val sspToTaskCount: mutable.Map[SystemStreamPartition, Int] = mutable.Map[SystemStreamPartition, Int]()
+    for(taskModel <- containerModel.getTasks.values())  {
+      for (systemStreamPartition <- taskModel.getSystemStreamPartitions) {
+        sspToTaskCount.put(systemStreamPartition, sspToTaskCount.getOrElse(systemStreamPartition, 0) + 1)
+      }
+    }
+
+    val chooser = DefaultChooser(inputStreamMetadata, chooserFactory, config, samzaContainerMetrics.registry, sspToTaskCount.toMap)
```

Comment:
Had an offline discussion. The consensus is passing in TaskCount map into the chooser will work, but it breaks the abstraction barrier and making the chooser aware of the number of downstream tasks that are consuming from it in unnecessary(violates the separation of concerns). 

The solution we'd agreed upon is retaining the lowest offset for each of the registered SystemStreamPartition and calling checkOffset for each SSP with lowest offset in BootstrappingChooser.start .